### ISSUE TITLE
Support for 'at' ufunc method

### ISSUE THREAD

**Issue by purepani** [pydata/sparse#548]
**Is your feature request related to a problem? Please describe.**
np.add.at(a, mask, b) is a replacement for a[mask] += b, except repeated indices will cause the addition to be performed as many times as they appear as opposed to just once. This ufunc method would be useful to support as using a for loop for this would be very costly.

**Describe the solution you'd like**
Add support for the at ufunc; currently a NotImplemented error is thrown when attempting to use np.add.at on sparse arrays.

**Comment by jakevdp** [comment]
Thanks for the request. JAX currently has an API for doing indexed updates that you can read about here: https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.ndarray.at.html#jax.numpy.ndarray.at

In brief, rather than `a[mask] += b` you could do `a_updated = a.at[mask].add(b)`.

Regarding the more familiar `ufunc.at` API, I have a work in progress implementing that here: https://github.com/google/jax/pull/9529. I hope to land that change soon.

All of the above discussion only concerns dense arrays; I'm not certain when we'll have this sort of syntax available for sparse arrays, as it poses a number of challenges within the current experimental implementation, namely concerning the need for knowing the size of the underlying sparse representation at compile-time.

**Comment by hameerabbasi** [comment]
@jakevdp I think you misunderstood the repo this issue was raised in. ;)

**Comment by jakevdp** [comment]
Ha, oops! Thanks.
